Userland/Services/WindowServer/Compositor.cpp

```diff
@@ -1160,7 +1168,16 @@ void Compositor::recompute_overlay_rects()
             return IterationDecision::Continue;
         });
 
-        invalidate_screen(render_rect);
+        if (was_invalidated) {
+            invalidate_screen(render_rect);
+        }
```

Comment:
Single line `if` block can do without the braces; same for the `for` loop a couple lines down.